### ISSUE TITLE
Error: Failed to execute 'postMessage' on 'Worker': ArrayBuffer at index 0 is already detached when using `worker: false`

### ISSUE THREAD

**Issue by semolex** [onthegomap/maplibre-contour#161]
### Description:
First of all - thank you for this library, it is really a great step to reduce number of unnecessary files.
I've been utilizing the `mlcontour` library in an Electron and Angular environment and have encountered some issues when using it with Maplibre GL.

Issue 1: Error with worker: true

When I set the worker option to `true` in the `mlcontour.DemSource` configuration, I receive the following error:

```node_modules_babel_runtime_helpers_esm_asyncToGenerator_js__WEBPACK_IMPORTED_MODULE_0__ is not defined```
This subsequently leads to a timeout error. Here is the configuration snippet where this issue occurs:

```javascript
const demSource = new mlcontour.DemSource({
  url: URL,
  encoding: "mapbox",
  maxzoom: 12,
  worker: true,
  cacheSize: 200,
  timeoutMs: 10_000,
});
```
### Issue 2: Error with worker: false

Alternatively, when I set the worker option to false, I encounter a different error:

```
Error: Failed to execute 'postMessage' on 'Worker': ArrayBuffer at index 0 is already detached
```
Here is the relevant configuration for this scenario:

```javascript
const demSource = new mlcontour.DemSource({
  url: URL,
  encoding: "mapbox",
  maxzoom: 12,
  worker: false,
  cacheSize: 200,
  timeoutMs: 10_000,
});
```
In this case contours are rendered, but sometimes area is empty.
### Environment:

Electron version: 23.1.4
Angular version: 15.2.0
Maplibre GL version: 3.5.2
Steps to Reproduce:

Initialize a new mlcontour.DemSource with the given configuration.
Observe the errors as mentioned above in the respective scenarios.

### Additional Information:
Tiles are served locally from mbtiles DEM source.
I tried to share protocol with other sources and seems like it reduces number of `detached` errors a little bit.
Any guidance or fixes for these issues would be greatly appreciated. I am eager to fully utilize this library in the Electron and Angular environment.

**Comment by etnav** [comment]
Do you have any updates on this issue?

Getting pretty much the same messages.

**Comment by msbarry** [comment]
The maplibre-contour build output uses async/await which is supported across all major browsers https://caniuse.com/async-functions. But, angular's build transpiles async/await using a library function that exists on the main thread, but not in the worker leading to errors like this. We had the same issue in maplibre (https://github.com/maplibre/maplibre-gl-js/issues/3209) and they ended up making the build target ES2016 so it wouldn't include async/await.

**Comment by msbarry** [comment]
@birkskyum I lost track of the maplibre fix after the initial change I made, did you end up finding any other workarounds for this?

**Comment by birkskyum** [comment]
We ended up lowering the output target to compile away the await/async.

**Comment by msbarry** [comment]
OK I lowered the target to es2015 and published 0.0.7, let me know if that works for you.

**Comment by semolex** [comment]
@msbarry Sorry for the late response, it works now with MapLibre v4. Thank you!
